packages/vite/src/node/build.ts

```diff
@@ -338,6 +338,22 @@ export function resolveBuildOptions(
         : defaultModulePreload
   }
 
+  // Auto-resolve dependencies if not specified
+  if (raw && raw.lib !== false && raw.rollupOptions?.external == null) {
+    const formats = raw.lib?.formats ?? []
+    const isUMD = formats.includes('umd')
+    const isIIFE = formats.includes('iife')
+    if (!isUMD && !isIIFE) {
+      const configRoot = process.cwd()
+      const pkg = lookupFile(configRoot, ['package.json'])
+      const dependencies: string[] =
+        (!!pkg && JSON.parse(pkg)?.dependencies) || []
+      if (dependencies.length > 0) {
+        resolved.rollupOptions.external = dependencies
```

Comment:
This was inspired by
https://github.com/vitejs/vite/blob/ee7c28a46a6563d54b828af42570c55f16b15d2c/packages/vite/src/node/config.ts#L902-L903

I have no clue yet if this works as intended, need to test it somehow
Also in contrast to `config.ts` I'm not calling `lookupFile` with a dynamic `configRoot` but always `process.cwd()` and I'm not sure if this what we want at the end